src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -283,124 +285,100 @@ XrdProtocol *XrdHttpProtocol::Match(XrdLink *lp) {
-      char *lnpos = strstr(SecEntity.moninfo, "/CN=");
-      char bufname2[9];
-      
-      
-      if (lnpos) {
-        lnpos += 4;
-        char *lnpos2 = index(lnpos, '/');
-        if (lnpos2) {
-          int l = ( lnpos2-lnpos < (int)sizeof(bufname) ? lnpos2-lnpos : (int)sizeof(bufname)-1 );
-          strncpy(bufname, lnpos, l);
-          bufname[l] = '\0';
-          
-          // Here we have the string in the buffer. Take the last 8 non-space characters
-          size_t j = 8;
-          strcpy(bufname2, "unknown-\0"); // note it's 9 chars
-          for (int i = (int)strlen(bufname)-1; i >= 0; i--) {
-            if (isalnum(bufname[i])) {
-              j--;
-              bufname2[j] = bufname[i];
-              if (j == 0) break;
-            }
-            
+
+  }
+
+  if (!SecEntity.name) {
+    // Here we have the user DN, and try to extract an useful user name from it
+    if (SecEntity.name) free(SecEntity.name);
+    SecEntity.name = 0;
+    // To set the name we pick the first CN of the certificate subject
+    // and hope that it makes some sense, it usually does
+    char *lnpos = strstr(SecEntity.moninfo, "/CN=");
+    char bufname2[9];
+
+
+    if (lnpos) {
+      lnpos += 4;
+      char *lnpos2 = index(lnpos, '/');
+      if (lnpos2) {
+        int l = ( lnpos2-lnpos < (int)sizeof(bufname) ? lnpos2-lnpos : (int)sizeof(bufname)-1 );
+        strncpy(bufname, lnpos, l);
+        bufname[l] = '\0';
+
+        // Here we have the string in the buffer. Take the last 8 non-space characters
+        size_t j = 8;
+        strcpy(bufname2, "unknown-\0"); // note it's 9 chars
+        for (int i = (int)strlen(bufname)-1; i >= 0; i--) {
+          if (isalnum(bufname[i])) {
+            j--;
+            bufname2[j] = bufname[i];
+            if (j == 0) break;
           }
-          
-          SecEntity.name = strdup(bufname);
-          TRACEI(DEBUG, " Setting link name: '" << bufname2+j << "'");
-          lp->setID(bufname2+j, 0);
-        }
-      }
-    }
-    
-    
-    // If we could not find anything good, take the last 8 non-space characters of the main subject
-    if (!SecEntity.name) {
-      size_t j = 8;
-      SecEntity.name = strdup("unknown-\0"); // note it's 9 chars
-      for (int i = (int)strlen(SecEntity.moninfo)-1; i >= 0; i--) {
-        if (isalnum(SecEntity.moninfo[i])) {
-          j--;
-          SecEntity.name[j] = SecEntity.moninfo[i];
-          if (j == 0) break;
-         
+
         }
+
+        SecEntity.name = strdup(bufname);
+        TRACEI(DEBUG, " Setting link name: '" << bufname2+j << "'");
+        lp->setID(bufname2+j, 0);
       }
     }
-   
-    
-    
   }
-  else return 0; // Don't fail if no cert
 
-  if (peer_cert) X509_free(peer_cert);
 
+  // If we could not find anything good, take the last 8 non-space characters of the main subject
+  if (!SecEntity.name) {
+    size_t j = 8;
+    SecEntity.name = strdup("unknown-\0"); // note it's 9 chars
```

Comment:
Is it safe to automatically assign names here (and above)?

Recall that parts of the DN are user-controlled ... not sure if a clever user could control their mapped name.

What if an admin wants a dedicated "unmapped" account when the user is not in the gridmap?

Reply:
This is probably a better question for @ffurano.

Reply:
I'm not able to answer about the "clever user" mapping an "unknown-*" name to something, maybe in a badly configured site, as I assume that the machine is safe.
Seems still reasonable to me to fill that field in that way if any other attempt to get a name fails. This applies also to the unencrypted case

Reply:
Ok - I think I understand this code better now.  I didn't see before that it always had a prefix of `unknown-`.